charts/vector/templates/_pod.tpl

```diff
@@ -174,6 +174,13 @@ tolerations:
 {{- end }}
 {{- with  .Values.topologySpreadConstraints }}
 topologySpreadConstraints:
+{{- range $_, $entry := . }}
+{{- if not (dig "labelSelectors" "matchLabels" false $entry) }}
+  {{- $ls := "labelSelectors:\n  matchLabels: {}" | fromYaml }}
```

Comment:
Not very familiar with this, but the `\n` seems a bit hacky. I wonder if we can do this more cleanly with a `dict`.

Reply:
I agree, it's kind of gross. I just updated this to use the `dict` function to create the nested dictionary keys.